langchain/document_loaders/confluence.py

```diff
@@ -0,0 +1,407 @@
+        if errors:
+            raise ValueError(f"Error(s) while validating input: {errors}")
+
+        self.base_url = url
+
+        try:
+            from atlassian import Confluence  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "`atlassian` package not found, please run"
+                "`pip install atlassian-python-api`"
+            )
+
+        if oauth2:
+            self.confluence = Confluence(url=url, oauth2=oauth2, cloud=cloud)
+        else:
+            self.confluence = Confluence(
+                url=url, username=username, password=api_key, cloud=cloud
+            )
+
+    @staticmethod
+    def validate_init_args(
+        url: Optional[str] = None,
+        api_key: Optional[str] = None,
+        username: Optional[str] = None,
+        oauth2: Optional[dict] = None,
+    ) -> Union[List, None]:
+        """Validates proper combinations of init arguments"""
+
+        errors = []
+        if url is None:
+            errors.append("Must provide `base_url`")
+
+        if (api_key and not username) or (username and not api_key):
+            errors.append(
+                "If one of `api_key` or `username` is provided,"
+                "the other must be as well."
+            )
+
+        if (api_key or username) and oauth2:
+            errors.append(
+                "Cannot provide a value for `api_key` and/or"
+                "`username` and provide a value for `oauth2`"
+            )
+
+        if oauth2 and oauth2.keys() != [
+            "access_token",
+            "access_token_secret",
+            "consumer_key",
+            "key_cert",
+        ]:
+            errors.append(
+                "You have either ommited require keys or added extra"
+                "keys to the oauth2 dictionary. key values should be"
+                "`['access_token', 'access_token_secret', 'consumer_key', 'key_cert']`"
+            )
+
+        if errors:
+            return errors
+        return None
+
+    def load(
+        self,
+        space_key: Optional[str] = None,
+        page_ids: Optional[List[str]] = None,
+        label: Optional[str] = None,
+        cql: Optional[str] = None,
+        include_attachments: bool = False,
+        limit: int = 50,
+    ) -> List[Document]:
+        """
+        :param space_key: Space key retrieved from a confluence URL, defaults to None
+        :type space_key: Optional[str], optional
+        :param page_ids: List of specific page IDs to load, defaults to None
+        :type page_ids: Optional[List[str]], optional
+        :param label: Get all pages with this label, defaults to None
+        :type label: Optional[str], optional
+        :param cql: CQL Expression, defaults to None
+        :type cql: Optional[str], optional
+        :param include_attachments: defaults to False
+        :type include_attachments: bool, optional
+        :param limit: Maximum number of pages to retrieve, defaults to 50
```

Comment:
thanks for this comment! at the very least we should document

Reply:
@zywilliamli @hwchase17  I opted to include pagination. Should be good now. I'll eventually open a PR to the Atlassian Python repo to support sending the raw response for all methods, because they only return the results key. This prevents us from using the `next` key provided by the API to check for additional results, and forces us to keep a counter relative to the limit.